### PR TITLE
Items page buttons

### LINKED ISSUE #115: Add Functionality to Buttons on Items Page
**User Story**
User story goes here

**Description**

**Acceptance Criteria**

Steps:

- step 1
- step 2

**Additional Information**

### PR DESCRIPTION
**Pull request by SilveerDusk**
## Developer: Jason Jelincic

Closes #115 

### Pull Request Summary

- Adds functionality to the delete item button
- Adds functionality to the view more item button
- Users can now edit their items on the items page
- Users can move their items from on basket to another
- Users can Create New Items

### Modifications

idk

### Testing Considerations

- Ran on local host

### Pull Request Checklist

- [ ] Code is neat, readable, and works
- [ ] Comments are appropriate
- [ ] The commit messages follows our [guidelines](https://h4i.notion.site/Conventional-Commits-593452ad1179489399ad3bd696ef772a)
- [ ] The developer name is specified
- [ ] The summary is completed
- [ ] Assign reviewers

### Screenshots/Screencast

Screenshot 2024-05-26 at 1 14 44 AM
Screenshot 2024-05-26 at 1 14 54 AM
Screenshot 2024-05-26 at 1 15 06 AM

### CODE REVIEW

**Comment by SilveerDusk** [comment]
Should I remove the compact group component if we plan on using Zac's Edit Item component? Figure we would use his because it has edit and delete item built into it. Also if we have the delete item button is it redundant to have it in the edit item component as well?

**Comment by zmattes04** [comment]
up to you i used that compact group as model to make the item one so maybe keep it for now if we need it still?

**Review by jhcsdev — APPROVED** [review]
Checked diff on all the files, looks good to me, don't have anything to add

**Comment by sofijadimitrijevic1** [comment]
Looks good to merge!

**Comment by SilveerDusk** [comment]
Screenshot 2024-05-26 at 12 35 53 PM
Screenshot 2024-05-26 at 12 36 08 PM

You can now move items!

**Comment by SilveerDusk** [comment]
Theres' a couple of weird bugs, Items can be doubled some times? And every action requires a page refresh cause adding anything to the useeffect causes an infinite loop

**Comment by SilveerDusk** [comment]
video preview on the group chat, cause its too big for github

**Comment by SilveerDusk** [comment]
Im thinking of turning the add item button into an item group to make it look cooler.
Screenshot 2024-05-26 at 2 28 47 PM
^ the current look
Also we need to start enforcing no more than one basket per item cause that causes most of the problems as well as only on group per basket.

**Comment by SilveerDusk** [comment]
Lowkey I want to merge this into main so we can keep going, but also keep it open so that I can keep working on it.

**Comment by SilveerDusk** [comment]
Also can we clear the database to like 3 users, 2 groups, 3-4 baskets, and 5 items? And update the schemas so that baskets have a group Id and the groups name and that items have the baskets name? It would help with mapping them all out as values.

I suggest: We keep Jelly Jason, g g, cow g, make a group for Jellly g and both g's, a basket for each user and one or two shared ones, and a handful of items.

For Example:
image
If items had the baskets name we could map it here without having to match the items to their basket with another fetch request.

**Comment by jhcsdev** [comment]
> Theres' a couple of weird bugs, Items can be doubled some times? And every action requires a page refresh cause adding anything to the useeffect causes an infinite loop

I think the group page has the same issue, I wonder if we can get around this by updating the frontend state with the returned post object, causing a re-render?

**Comment by jhcsdev** [comment]
> Im thinking of turning the add item button into an item group to make it look cooler. Screenshot 2024-05-26 at 2 28 47 PM ^ the current look Also we need to start enforcing no more than one basket per item cause that causes most of the problems as well as only on group per basket.

Not sure what that would entail but if it isn't too difficult I say go for it

**Comment by jhcsdev** [comment]
> Lowkey I want to merge this into main so we can keep going, but also keep it open so that I can keep working on it.

Down

**Comment by jhcsdev** [comment]
> Also can we clear the database to like 3 users, 2 groups, 3-4 baskets, and 5 items? And update the schemas so that baskets have a group Id and the groups name and that items have the baskets name? It would help with mapping them all out as values.
> 
> I suggest: We keep Jelly Jason, g g, cow g, make a group for Jellly g and both g's, a basket for each user and one or two shared ones, and a handful of items.
> 
> For Example: image If items had the baskets name we could map it here without having to match the items to their basket with another fetch request.

Sure that sounds good to me. I don't think there's any reason to keep it all in so I say send it

File: frontend/src/components/EditItem.tsx
Comment (jhcsdev) [inline frontend/src/components/EditItem.tsx:1]: where'd the delete option go? is it somewhere else now

Comment (jhcsdev) [reply frontend/src/components/EditItem.tsx:1]: Nvm, just read thru ItemGroup.tsx

**Review by jhcsdev — APPROVED** [review]
Good cleanup and refactoring going on nice work jdawg

**Pull request merged**